frame/support/procedural/src/benchmark.rs

```diff
@@ -152,6 +173,41 @@ struct BenchmarkDef {
 impl BenchmarkDef {
 	/// Constructs a [`BenchmarkDef`] by traversing an existing [`ItemFn`] node.
 	pub fn from(item_fn: &ItemFn, extra: bool, skip_meta: bool) -> Result<BenchmarkDef> {
+		let invalid_return = |span| {
+			return Err(Error::new(
+				span,
+				"Invalid return type. Only `()` and `Result<(), BenchmarkError>` are allowed.",
+			))
+		};
+		if let ReturnType::Type(_, typ) = &item_fn.sig.output {
+			// parsing for non `()` return type (only allow `Result<(), BenchmarkError>`)
```

Comment:
Could you please put all this added code in its own function? Like `parse_return_type` or whatever.

Reply:
yeah I can pull this out.